src/vscode-bicep/src/commands/deploy.ts

```diff
@@ -51,6 +52,15 @@ export class DeployCommand implements Command {
     context: IActionContext,
     documentUri: vscode.Uri | undefined
   ): Promise<void> {
+    const requestId = Math.random().toString();
+
+    await callWithTelemetryAndErrorHandling(
```

Comment:
I'm now thinking it would be better to actually send deploy/start from the language server, so that if called from VS or other editor we'll still get start/end events.  The deployment command still has an automatic telemetry event from the extension.

Reply:
In which case you can generate the id in the lang server.  Recommend using a random guid.

Reply:
Hmm I did mention this option during one of our meetings and @ucheNkadiCode wanted to capture time between right-click deploy and deployment complete. 

We do some work before we hit the language server layer - build to find validation issues, figure out target scope, capture data from user like parameter file path, location, resource group creation (in case of rg level deployment). Do we not want to include that time in our telemetry data?

Also, the default telemetry event from the deployment command shows up after deployment is complete. "deploy/result" event from language server shows up before the one from the command. So it might not be useful in deployment duration calculation, unless we want to capture start and end in vscode layer(deploy.ts).

Reply:
With all due respect to Uche, the important measurement is deployment start vs stop (what if they invoke deploy, then go off for a break before finishing the inputs?).  Uche can still get his full time either way by subtracting the commands/bicep.deploy event's time from the result event.

Reply:
As discussed offline, moved the deploy/start to language server layer. I have included the deployId in the default event that gets triggered by the deploy command.

https://github.com/Azure/bicep/commit/f966ce69fc2547ef8a610340e8cde9a100e5ca0b

Thanks!